openfisca_survey_manager/simulations.py

```diff
@@ -0,0 +1,1271 @@
+    elif aggfunc == 'count':
+        aggregate = (weight * filter_dummy).sum()
+    elif aggfunc == 'count_non_zero':
+        aggregate = (weight * (value != 0) * filter_dummy).sum()
+    else:
+        aggregate = None
+
+    return aggregate
+
+
+def compute_quantiles(simulation: Simulation, variable: str, nquantiles: int = None,
+        period: Optional[Union[int, str, Period]] = None, filter_by = None, weighted: bool = True,
+        alternative_weights = None, filtering_variable_by_entity = None) -> List[float]:
+    """
+    Compute quantiles of a variable.
+
+    Args:
+        simulation (Simulation, optional): Simulation to be used. Defaults to None.
+        variable (str, optional): Variable which quantiles are computed. Defaults to None.
+        nquantiles (int, optional): Number of quantiles. Defaults to None.
+        period (Optional[Union[int, str, Period]], optional): Period. Defaults to None.
+        missing_variable_default_value (optional): Value to use for missing values. Defaults to np.nan.
+        weighted (bool, optional): Whether to weight the variable or not. Defaults to True.
+        alternative_weights (Optional[Union[str, int, float, Array]], optional): Alternative weigh to use. Defaults to None.
+        filtering_variable_by_entity (Dict, optional): Filtering variable by entity. Defaults to None.
+
+    Returns:
+       List(float): The quantiles values
+    """
+    weight_variable_by_entity = simulation.weight_variable_by_entity
+    weight_variable = None
+    entity_key = simulation.tax_benefit_system.variables[variable].entity.key
+    if weight_variable_by_entity:
+        weight_variable = weight_variable_by_entity[entity_key]
+
+    variable_values = simulation.adaptative_calculate_variable(variable, period)
+    if weighted:
+        assert (alternative_weights is not None) or (weight_variable is not None)
+        weight = (
+            alternative_weights
+            if alternative_weights is not None
+            else simulation.calculate(weight_variable, period)
+            )
+    else:
+        weight = np.ones(len(variable_values))
+
+    if filtering_variable_by_entity is not None:
+        if filter_by is None:
+            filter_by = filtering_variable_by_entity.get(entity_key)
+
+    if filter_by is not None:
+        filter_entity_key = simulation.tax_benefit_system.variables.get(filter_by).entity.key
+        assert filter_entity_key == entity_key
+        filter_dummy = simulation.calculate(filter_by, period = period).astype(bool)
+
+        variable_values = variable_values[filter_dummy].copy()
+        weight = weight[filter_dummy].copy()
+
+    labels = np.arange(1, nquantiles + 1)
+    method = 2
+    _, values = mark_weighted_percentiles(variable_values, labels, weight, method, return_quantiles = True)
+    return values
+
+
+def compute_pivot_table(simulation: Simulation = None, baseline_simulation: Simulation = None, aggfunc = 'mean',
```

Comment:
Because we have to combine `simulation` and `baseline_simulation` at the micro-data level?.

Reply:
Because we have to combine `simulation` and `baseline_simulation` at the micro-data level ?.
For example we may want some mean of difference of some values (let's say `ompot_revenu` for France) within cells of another variable (as `revenu_disponible` or `rfr`)